parser/feed/src/main/java/de/danoeh/antennapod/parser/feed/PodcastIndexTranscriptParser.java

```diff
@@ -0,0 +1,191 @@
+                    if (timecodes.length < 2) {
+                        continue;
+                    }
+                    startTimecode = parseTimecode(timecodes[0].trim());
+                    endTimecode = parseTimecode(timecodes[1].trim());
+                    if (spanStartTimecode == 0) {
+                        spanStartTimecode = startTimecode;
+                    }
+                    duration += endTimecode - startTimecode;
+                    do {
+                        line = iter.next();
+                        if (StringUtil.isBlank(line)) {
+                            break;
+                        }
+                        body.append(line.strip());
+                        body.append(" ");
+                    } while (iter.hasNext());
+                }
+
+                if (body.indexOf(":") != -1) {
+                    String [] parts = body.toString().split(":");
+                    speaker = parts[0];
+                    body = new StringBuffer(parts[1].strip());
+                }
+                if (!StringUtil.isBlank(body.toString())) {
+                    segmentBody += " " + body;
+                    segmentBody = StringUtils.trim(segmentBody);
+                    if (duration >= span && endTimecode > spanStartTimecode) {
+                        Log.d(TAG, "SRT : " + Long.toString(spanStartTimecode) + " " + segmentBody);
+                        transcript.addSegment(new TranscriptSegment(spanStartTimecode,
+                                endTimecode,
+                                segmentBody,
+                                speaker));
+                        duration = 0L;
+                        spanStartTimecode = 0L;
+                        segmentBody = "";
+                    }
+                    body = new StringBuffer("");
+                }
+            } while (iter.hasNext());
+
+            if (! StringUtil.isBlank(segmentBody) && endTimecode > spanStartTimecode) {
+                Log.d(TAG, "SRT : " + Long.toString(spanStartTimecode) + " " + segmentBody);
+
+                segmentBody = StringUtils.trim(segmentBody);
+                transcript.addSegment(new TranscriptSegment(spanStartTimecode,
+                        endTimecode,
+                        segmentBody,
+                        speaker));
+            }
+            if (transcript.getSegmentCount() > 0) {
+                transcript.setRawString(str);
+                return transcript;
+            } else {
+                return null;
+            }
+        }
+
+        static long parseTimecode(String timecode) {
+            String[] parts = timecode.split(":");
+            try {
+                int hours = Integer.parseInt(parts[0]);
+                int minutes = Integer.parseInt(parts[1]);
+                int seconds = Integer.parseInt(parts[2].substring(0, 2));
+                int milliseconds = Integer.parseInt(parts[2].substring(3));
```

Comment:
Doesn't the access to the array crash when the input is malformed?

Reply:
thanks, added some protection in front for length of `parts`

Reply:
Maybe a regex could also be used here to make sure that it follows the right format

Reply:
done, using a regex now